### ISSUE TITLE
Improve wfs_api() error reporting

### ISSUE THREAD

**Issue by jlehtoma** [rOpenGov/geofi#19]
Currently `wfs_api()` passes the provided WFS URL as-is and expects a correct response. However, it is not robust against faulty URLs (404), timeouts (408) or any other unexpected situations. It would be good if it did. In principle this would be up to `ows4R`, but nut sure at this point how it deals with errors.

**Comment by jlehtoma** [comment]
Since we're not using `ows4R`, the checking responsibility is now on us (and maybe it was before too).

**Comment by jlehtoma** [comment]
@muuankarski we might be able to replace [`check_api_access()`](https://github.com/rOpenGov/geofi/blob/master/R/check_api_access.R) with little enhancements in `wfs_api()`. If you can figure out which [HTTP status code](https://www.restapitutorial.com/httpstatuscodes.html) you're receiving while behind company firewall / proxy, we could include warnings to the user [here](https://github.com/rOpenGov/geofi/blob/master/R/wfs_api.R#L69). Could you test what `httr::GET()` will give you (see [here](https://github.com/rOpenGov/geofi/blob/master/R/wfs_api.R#L59) onwards) in case there's no access?

**Comment by muuankarski** [comment]
@jlehtoma 

as we got access to geo.stat.fi from our servers yesterday, I will try to reach a layer in Paituli wfs.

```
# Desktop windows
httr::http_status(httr::GET("http://avaa.tdata.fi/geoserver/paituli/wfs?service=WFS&version=2.0.0&request=getFeature&typename=paituli:kotus_pitajaAll"))
Error in curl::curl_fetch_memory(url, handle = handle) : 
  Timeout was reached: [avaa.tdata.fi] Connection timed out after 10000 milliseconds```
# server
> httr::http_status(httr::GET("http://avaa.tdata.fi/geoserver/paituli/wfs?service=WFS&version=2.0.0&request=getFeature&typename=paituli:kotus_pitajaAll"))
$category
[1] "Client error"

$reason
[1] "Forbidden"

$message
[1] "Client error: (403) Forbidden"
```

So the problem with proxy requiring authentication is that you get no http status code from there. Therefore this `download.file()`-hack. But there must be more elegant way to address this?

**Comment by jlehtoma** [comment]
Right! Thanks for checking this. I had a quick look at `httr`/`curl` and it seems that although it is possible to pass a custom failure callback to `curl::curl_fetch_memory()` used by `httr::GET()`, we can't actually pass such a callback through ``httr::GET()`. Two options come to mind:

1. Do a separate probe with `curl::curl_fetch_memory()` (much like with `download.file()` now) within `wfs_api()` using a custom failure callback to let user know about potential issue with a proxy.
2. Use `tryCatch()` with `GET()` within `wfs_api()`, catch timeout error (by parsing the error message) and let user know about potential issue with a proxy.
